pandas/core/generic.py

```diff
@@ -1787,6 +1787,10 @@ def _slice(self, slobj, axis=0, kind=None):
         return result
 
     def _set_item(self, key, value):
+        if callable(getattr(self, key, None)):
+            warnings.warn("Pandas doesn't allow attribute-like access to "
+                          "columns whose names collide with methods",
```

Comment:
i would say that this is a collision

we actually DO allow it (maybe should raise but let's start with a warning)